### PR TITLE
[command] create a complete lexer

### LINKED ISSUE #205: Improve Sub Command parsing and cacheing
We will want to lexer it

Something like this
```java
public enum CommandTokenType {
    KEYWORD,
    ARGUMENT_PARSER
}
```

```java
public interface CommandToken {
    CommandTokenType getType();
}
```

Also #105 should be added because that how the new lexer work

### PR DESCRIPTION
**Pull request by ikfir**
### ❓ Type of change

- [X] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (adds functionality)

### 📚 Description
Lexer (Lexical analysis) is conversion of a text into (semantically or syntactically) meaningful lexical tokens belonging to categories defined by a "lexer" program,
For our use cases it would be used for `@SubCommand` `@CompoundParser` `@FlagVarint` etc...

Closes #205 

### 🧪 How Has This Been Tested?
a lot of unit test

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

## Release Notes for Version 0.0.37

- **New Features**
  - Enhanced command handling with improved serialization of subcommands.
  - Introduced a new `CommandLexer` interface for better command tokenization.
  - Added support for parameter names in argument parsers, improving clarity.
  - New `UnknownTokenException` for handling unknown command tokens during parsing.

- **Bug Fixes**
  - Updated exception handling in command tests for more precise error reporting.

- **Documentation**
  - Updated changelog to reflect new features and enhancements.

- **Tests**
  - Added comprehensive unit tests for the new `SimpleCommandLexer` functionality, ensuring robust error handling and tokenization.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
 # CHANGELOG
+## 0.0.37
+- [commands] Improve SubCommand serialization with Command Lexer
+- [commands] Add Parameter names for argument parsers
```
Comment (liorsl) [inline CHANGELOG.md:4]: didn't you say this will be in another pr?

Comment (ikfir) [reply CHANGELOG.md:4]: opss

Comment (ikfir) [reply CHANGELOG.md:4]: but it's true
